modules/snooker/src/main/ts/ephox/snooker/selection/TableSelection.ts

```diff
@@ -0,0 +1,34 @@
+import { Element as DomElement, HTMLTableCellElement, HTMLTableRowElement, HTMLTableCaptionElement, Node } from '@ephox/dom-globals';
+import { Arr, Options } from '@ephox/katamari';
+import { TableLookup } from '@ephox/snooker';
+import { Attr, Compare, Element, Elements, SelectorFind } from '@ephox/sugar';
+import { Selections } from './Selections';
+import CellOperations from '../queries/CellOperations';
+
+const getSelectionStartFromSelector = <T extends DomElement>(selector: string) => (start: Element<Node>) =>
+  SelectorFind.closest<T>(start, selector);
+
+const getSelectionStartCaption = getSelectionStartFromSelector<HTMLTableCaptionElement>('caption');
+
+const getSelectionStartCell = getSelectionStartFromSelector<HTMLTableCellElement>('th,td');
+
+const getSelectionStartCellOrCaption = getSelectionStartFromSelector<HTMLTableCellElement | HTMLTableCaptionElement>('th,td,caption');
+
+const getCellsFromSelection = (body: any, start: Element<Node>, selectedSelector: string): Element<HTMLTableCellElement>[] =>
```

Comment:
Body should have a type here. It would probably also make more sense to call it `root` or something similar.